demostuff

```diff
@@ -27,7 +27,9 @@ and defines gapp, window, box, button, text, xtext, and xlabel.
             are the button and the new text.
 |#
 
-(load-shared-object "/usr/lib/x86_64-linux-gnu/libgtk-3.so.0")
+(require "asdf")
+
+(load-shared-object (uiop:run-program "ldconfig -p | grep libgtk-3.so.0 | tr ' ' '\\n' | grep /" :output '(:string :stripped t)))
```

Comment:
If you want to avoid using ASDF, you can use `sb-ext:run-program` to get the output of `ldconfig -p`.

Pipes don't seem to work with `sb-ext:run-program`, so the output would have to be explicitly captured and passed.

Playing around with it, I got this to work:
```
(load-shared-object
  (let ((to-process (with-output-to-string (gtk-line)
                      (with-input-from-string (to-grep (with-output-to-string (libraries)
                                                         (sb-ext:run-program "/usr/bin/ldconfig"
                                                                             (list "-p")
                                                                             :output libraries)))
                        (sb-ext:run-program "/usr/bin/grep"
                                            (list "libgtk-3.so.0" "-")
                                            :input to-grep
                                            :output gtk-line)))))
   (string-trim '(#\Space #\Newline) (subseq to-process (position #\Space to-process :from-end 't))))
```

I'm sure there are cleaner ways to do it, but it'll get the job done if not having dependencies is a priority.

Reply:
How about:

(load-shared-object
  (let ((s (with-output-to-string (o)
    (sb-ext:run-program "/bin/bash"
      (list "-c" "ldconfig -p | grep libgtk-3.so") :output o))))
    (subseq s 1 (position #\space s))))

Reply:
That code boils down to 
`(load-shared-object "libgtk-3.so.0")`

And the sbcl manual states (http://www.sbcl.org/manual/#Loading-Shared-Object-Files)

> Locating the shared object follows standard rules of the platform, consult the manual page for dlopen(3) for details. Typically paths specified by environment variables such as LD_LIBRARY_PATH are searched if the pathname has no directory...

So just using that should be fine on all systems.

Reply:
I changed it to (load-shared-object "libgtk-3.so.0")

Reply:
I'm closing this PR then. Great job guys!